params/extras/precompile_upgrade_test.go

```diff
@@ -277,16 +277,16 @@ type upgradeCompatibilityTest struct {
 func (tt *upgradeCompatibilityTest) run(t *testing.T, chainConfig ChainConfig) {
 	// apply all the upgrade bytes specified in order
 	for i, upgrade := range tt.configs {
-		newCfg := Copy(&chainConfig)
-		GetExtra(&newCfg).UpgradeConfig = *upgrade
+		newCfg := chainConfig
+		newCfg.UpgradeConfig = *upgrade
 
-		err := chainConfig.CheckCompatible(&newCfg, 0, tt.startTimestamps[i])
+		err := chainConfig.checkConfigCompatible(&newCfg, new(big.Int), tt.startTimestamps[i])
```

Comment:
why the change from `CheckCompatible` to `checkConfigCompatible`? Can't we keep it to `checkCompatible` (unexported)?

Reply:
`CheckConfigCompatible` is referred to in libvem, and `CheckConfigCompatible` converts types and calls the unexported `checkConfigCompatible`.
Since this is how it is in coreth & libevm, I'm not sure what we gain by changing it.